glide-core/redis-rs/redis/src/aio/mod.rs

```diff
@@ -105,8 +114,38 @@ impl Clone for Box<dyn DisconnectNotifier> {
     }
 }
 
+// Helper function to extract and update availability zone from INFO command
+async fn update_az_from_info<C>(con: &mut C) -> RedisResult<()>
+where
+    C: ConnectionLike,
+{
+    let info_res = con.req_packed_command(&cmd("INFO")).await;
+
+    match info_res {
+        Ok(value) => {
+            let info_dict: InfoDict = FromRedisValue::from_redis_value(&value)?;
+            if let Some(node_az) = info_dict.get::<String>("availability_zone") {
+                con.set_az(Some(node_az));
```

Comment:
I would also add a debug message stating that the property "availability_zone" could not be found for this redis server

Reply:
@eifrah-aws  I dont think its required - not having AZ is ok (redis/older versions)